File: backend/app/schemas/weather.py
from datetime import datetime

from pydantic import BaseModel


class WeatherSummary(BaseModel):
    label: str
    available: bool
    temperature: float | None = None
    temperature_min: float | None = None
    temperature_max: float | None = None
    precipitation_probability: int | None = None
    precipitation_hint: str = "derzeit nicht verfuegbar"
    wind_speed: float | None = None
    summary: str = "derzeit nicht verfuegbar"
    updated_at: datetime | None = None
    is_cached: bool = False

File: frontend/src/types/weather.ts
export type WeatherSummary = {
  label: string;
  available: boolean;
  temperature: number | null;
  temperature_min: number | null;
  temperature_max: number | null;
  precipitation_probability: number | null;
  precipitation_hint: string;
  wind_speed: number | null;
  summary: string;
  updated_at: string | null;
  is_cached: boolean;
};
